pages/admin/lessons/new/index.tsx

```diff
@@ -1,94 +1,91 @@
-import React, { useState } from 'react'
-import { gql, useLazyQuery } from '@apollo/client'
+import React from 'react'
+import _ from 'lodash'
 import { AdminLessonCard } from '../../../../components/admin/lessons/AdminLessonCard'
-import { withGetApp, GetAppProps } from '../../../../graphql'
+import {
+  withGetApp,
+  GetAppProps,
+  useGetFlaggedExercisesQuery,
+  GetFlaggedExercisesQuery
+} from '../../../../graphql'
 import { AdminLayout } from '../../../../components/admin/AdminLayout'
 import Link from 'next/link'
 import styles from '../../../../scss/adminLessonPage.module.scss'
 
-const EXERCISES = gql`
-  query {
-    exercises {
-      flaggedAt
-      module {
-        lesson {
-          title
-        }
-      }
-    }
-  }
-`
-type Exercise = {
-  flaggedAt?: string
-  module: {
-    lesson: {
-      title: string
-    }
-  }
+type Lesson = {
+  id: number
+  title: string
+  description: string
+  docUrl?: string | null
+  githubUrl?: string | null
+  videoUrl?: string | null
+  order: number
+  slug: string
+  chatUrl?: string | null
+  challenges: Array<{
+    id: number
+    title: string
+    description: string
+    order: number
+  }>
 }
 
-type Exercises = Exercise[]
-
-const Lessons: React.FC<GetAppProps> = ({ data }) => {
-  const { lessons } = data
-  const [lessonCards, setCards] = useState<[]>([])
-  const [getExercises, { data: exercisesData }] = useLazyQuery<{
-    exercises: Exercises
-  }>(EXERCISES, {
-    onCompleted: async () => {
-      const exerciseMapping = await exercisesData?.exercises.reduce(
-        (prev: any, cur) => {
-          if (cur.flaggedAt) {
-            if (!prev[cur.module?.lesson?.title]) {
-              prev[cur.module?.lesson?.title] = 1
-              return prev
-            }
-            prev[cur.module?.lesson?.title]++
-          }
+type LoadedLessonCardsProps = {
+  lessonsData: Lesson[] | undefined
+  exercisesData: GetFlaggedExercisesQuery | undefined
+}
 
-          return prev
-        },
-        {}
-      )
+const LoadedLessonCards = ({
+  lessonsData,
+  exercisesData
+}: LoadedLessonCardsProps) => {
+  const exerciseMapping = exercisesData?.exercises.reduce(
+    (acc: { [key: string]: number }, cur) => {
+      const curTitle = _.get(cur, 'module.lesson.title', undefined)
```

Comment:
Here `get` was used, but in line `41` `?.` was used.

Reply:
Im going to change both to `?. `optional chaining. Line 41 was giving me issues with lodash("`error: object is possibly undefined"`)

Reply:
> Im going to change both to `?. `optional chaining. Line 41 was giving me issues with lodash("`error: object is possibly undefined"`)

I think to solve it, you could do 
```ts
const exercises = _.get(exercisesData, 'exercises')

const exerciseMapping = exercises && exercises.reduce...
```

Reply:
@HS-90 i dont think its a problem with lodash, I think its a problem with how you set up your logic.

I initially speculated the problem might have been that lodash `_.get` did not cases where first parameter has value of undefined. But then I went to lodash.com and tried the following in my console:
```
a = undefined
b = _.get(a, 'ss', 'potato')
console.log(b) // logs potato
```

So it looks like _.get can handle when the first parameter has a value of undefined. You shouldn't have to do @flacial 's suggestion either of `exercises && exercises.reduce`. Thats an an anti pattern since we already have `_.get` to set default values for nested properties that are not found
If the problem is with `exercises` possibly being undefined, just set its default value as `[]`
```
const exercises = _.get(exercisesData, 'exercises', [])
```

Reply:
If you set the default value to empty array `[]` while using `reduce`, Typescript will throw an error https://github.com/microsoft/TypeScript/issues/36390#issuecomment-852533093 

However, `forEach` works in this case. So I set it up a little differently than before:

![image](https://user-images.githubusercontent.com/77421872/179883954-381b2e53-da08-4f62-954d-8a9fd9761ec3.png)